src/variorum/IBM/Power9.c

```diff
@@ -440,6 +441,42 @@ int ibm_cpu_p9_get_node_power_json(char **get_power_obj_str)
     return 0;
 }
 
+int ibm_cpu_p9_get_node_util_json(char **get_power_obj_str)
+{
+    char *val = ("VARIORUM_LOG");
+    if (val != NULL && atoi(val) == 1)
+    {
+        printf("Running %s\n", __FUNCTION__);
+    }
+
+    int ru;
+    char hostname[1024];
+    struct timeval tv;
+    struct rusage rusge;
+    uint64_t ts;
+    long mem;
+    json_t *get_util_obj = json_object();
+
+    gethostname(hostname, 1024);
+    gettimeofday(&tv, NULL);
+    ts = tv.tv_sec * (uint64_t)1000000 + tv.tv_usec;
+    json_object_set_new(get_util_obj, "host", json_string(hostname));
+    json_object_set_new(get_util_obj, "timestamp", json_integer(ts));
+
+    ru = getrusage(RUSAGE_SELF, &rusge);
+    if (ru < 0)
+    {
+        printf("Failed to get utilizations\n");
+        return -1;
+    }
+
+    mem = rusge.ru_maxrss;
```

Comment:
@altahat2003 
- The `getrusage` part will be common to all architectures, and we will have to maintain CPU and memory delta values. The GPU usage will be different based on which architecture you are on. My suggestion is that you move this code out to a higher level in `variorum.c` itself instead of creating redundant code in each lower-level platform file.

We have a `print_gpu_utilization` API, and we can add a parallel `get_gpu_utilization_json` API, that you can then call from `variorum.c` (your code on lines 1013 onward). You don't really need a platform-specific implementation for the CPU and memory usage part. Just for the GPU part.

Reply:
Summarizing the suggestion from our standup call here: Could the high-level utilization API be parameterized to choose between (a) reporting accumulator-style data when available on the underlying system or (b) reporting delta between current and previous sample? This will satisfy a variety of use cases depending on the monitoring requirements and configurations of the client application. For example, RoCM reports GPU utilization via. both accumulator-style utilization API as well as instantaneous utilization API. So the client could select which underlying API it wants to use.